src/game/option_passport.c

```diff
@@ -57,7 +78,84 @@ REQUEST_INFO g_LoadSavegameRequester = {
     0,
 };
 
-static void InitNewGameRequester()
+static void Option_PassportInitText();
+static void Option_PassportShutdownText();
+static void Option_PassportShowNewGame();
+static void Option_PassportShowSaves();
+static void Option_PassportInitText();
+static void Option_PassportInitNewGameRequester();
+static void Option_PassportInitSaveRequester();
+
+static void Option_PassportInitText()
+{
+    m_Text[TEXT_LEFT_ARROW] = Text_Create(-85, -15, "@");
+    Text_AlignBottom(m_Text[TEXT_LEFT_ARROW], 1);
+    Text_CentreH(m_Text[TEXT_LEFT_ARROW], 1);
+    Text_Hide(m_Text[TEXT_LEFT_ARROW], true);
+
+    m_Text[TEXT_RIGHT_ARROW] = Text_Create(70, -15, "\t");
+    Text_AlignBottom(m_Text[TEXT_RIGHT_ARROW], 1);
+    Text_CentreH(m_Text[TEXT_RIGHT_ARROW], 1);
+    Text_Hide(m_Text[TEXT_RIGHT_ARROW], true);
+
+    m_Text[TEXT_PAGE_NAME] = Text_Create(0, -16, "");
+
+    for (int i = 0; i < TEXT_NUMBER_OF; i++) {
+        Text_AlignBottom(m_Text[i], 1);
+        Text_CentreH(m_Text[i], 1);
+    }
+}
+
+static void Option_PassportShutdownText()
+{
+    for (int i = 0; i < TEXT_NUMBER_OF; i++) {
+        Text_Remove(m_Text[i]);
+        m_Text[i] = NULL;
+    }
+    m_PassportMode = PASSPORT_MODE_FLIP;
+    m_IsTextInit = false;
+}
+
+static void Option_PassportShowNewGame()
+{
+    int32_t select = DisplayRequester(&m_NewGameRequester);
+    if (select) {
+        if (select > 0) {
+            g_InvExtraData[1] = select - 1;
```

Comment:
Can you please remind me what does `DisplayRequester` return and what do we store in `g_InvExtraData`?

Reply:
`DisplayRequester ` returns -1 if the user deselects, 0 if the user makes no action, and the save game slot number if the user selects.

`g_InvExtraData` I don't know as well. It's an 8 length array that only seems to store two different things. Index `[0]` stores the passport page number, and index `[1]` stores the save game slot number.